### ISSUE TITLE
Ecommerce Sync throw a fatal error when First date is empty

### ISSUE THREAD

**Issue by cavalier79** [mailchimp/mc-magento2#1093]
## Issue Description

the cron process cant proceed to sync the ecommerce when the first date (Mailchimp -> configuration -> Ecommerce Configuration -> First Date) is empty (first install ad example).

This cause to generate a query like this:

```
SELECT 
	`main_table`.*, `m4m`.* 
FROM 
	`newsletter_subscriber` AS `main_table` 
	LEFT JOIN `defaultdb`.`mailchimp_sync_ecommerce` AS `m4m` ON m4m.related_id = main_table.subscriber_id and m4m.type = 'SUB' and m4m.mailchimp_store_id = 'XXXXXXXX' 
WHERE 
	(`subscriber_status` = 1) AND 
	(`main_table`.`store_id` = 1) AND 
	(m4m.mailchimp_sync_delta IS null OR (m4m.mailchimp_sync_delta > '' and m4m.mailchimp_sync_modified = 1))
```

This query throw this error:

/* Errore SQL (1525): Incorrect DATETIME value: '' */

Because the field m4m.mailchimp_sync_delta is a DATETIME field and mysql throws the exception.

The query is generated in vendors/mailchimp/mc-magento2/Model/Api/Product.php in rows 137:

```
$collection->getSelect()->where("m4m.mailchimp_sync_delta IS null OR (m4m.mailchimp_sync_delta > '".
            $this->_helper->getMCMinSyncDateFlag().
            "' and m4m.mailchimp_sync_modified = 1)");
        $collection->getSelect()->limit(self::MAX);
```

And should be fixed adding a CAST function around the $this->_helper->getMCMinSyncDateFlag()

```
$collection->getSelect()->where("m4m.mailchimp_sync_delta IS null OR (m4m.mailchimp_sync_delta > CAST('".
            $this->_helper->getMCMinSyncDateFlag().
            "' AS DATETIME) and m4m.mailchimp_sync_modified = 1)");
        $collection->getSelect()->limit(self::MAX);

```

Precondition: 

Magento 2.3.5.p1
Mc-Magento2 (<module name="Ebizmarts_MailChimp" setup_version="102.3.41">)
running on a droplet with Mysql 8.0.20

### Steps to reproduce

Leave the first date field in ecommerce configuration blank.

Using Amasty Scheduled Cron List or a similiar module or reading cron.log:

```
[2021-01-20 15:10:05] main.CRITICAL: SQLSTATE[HY000]: General error: 1525 Incorrect DATETIME value: '', query was: SELECT `main_table`.*, `m4m`.* FROM `newsletter_subscriber` AS `main_table`
 LEFT JOIN `defaultdb`.`mailchimp_sync_ecommerce` AS `m4m` ON m4m.related_id = main_table.subscriber_id and m4m.type = 'SUB' and m4m.mailchimp_store_id = 'XXX' WHERE (`subscriber_status` = 1) AND (`main_table`.`store_id` = 1) AND (m4m.mailchimp_sync_delta IS null OR (m4m.mailchimp_sync_delta > '' and m4m.mailchimp_sync_modified = 1)) LIMIT 100 {"exception":"[object] (Zend_Db_Statement_Exception(code: 0): SQLSTATE[HY000]: General error: 1525 Incorrect DATETIME value: '', query was: SELECT `main_table`.*, `m4m`.* FROM `newsletter_subscriber` AS `main_table`
 LEFT JOIN `defaultdb`.`mailchimp_sync_ecommerce` AS `m4m` ON m4m.related_id = main_table.subscriber_id and m4m.type = 'SUB' and m4m.mailchimp_store_id = 'XXX' WHERE (`subscriber_status` = 1) AND (`main_table`.`store_id` = 1) AND (m4m.mailchimp_sync_delta IS null OR (m4m.mailchimp_sync_delta > '' and m4m.mailchimp_sync_modified = 1)) LIMIT 100 at /home/xyz.com/public_html/vendor/magento/framework/DB/Statement/Pdo/Mysql.php:110, PDOException(code: HY000): SQLSTATE[HY000]: General error: 1525 Incorrect DATETIME value: '' at /home/xyz.com/public_html/vendor/magento/framework/DB/Statement/Pdo/Mysql.php:91)"} []
```

### Actual and Expected result

Actual result: no ecommerce sync
Expected result: ecommerce order synced with MC.

**Comment by gonzaloebiz** [comment]
Update the extension to the current version

Best

**Comment by cavalier79** [comment]
The issue is present in the current version.

   $collection->getSelect()->where("m4m.mailchimp_sync_delta IS null OR (m4m.mailchimp_sync_delta > **'".
            $this->_helper->getMCMinSyncDateFlag().
            "'** and m4m.mailchimp_sync_modified = 1)");
        $collection->getSelect()->limit(self::MAX);

Please check your code and read my entire issue text before closing.

**Comment by igormatkovic** [comment]
Can you please tag a new version with this fix? 

The latest tagged: 102.3.41 doesn't have this fix 

Thanks

**Comment by cavalier79** [comment]
Exactly what i said, probably i wrote in bad english or the MC team doesnt understand.


The issue is located HERE:

https://github.com/mailchimp/mc-magento2/blob/develop-2.3/Model/Api/Product.php

And obviously is presente also here:

https://github.com/mailchimp/mc-magento2/blob/102.3.41/Model/Api/Product.php

please someone REOPEN this issue and apply a fix.

**Comment by gonzaloebiz** [comment]
Take a look at https://github.com/mailchimp/mc-magento2/issues/1066.
This fix will be available in the next release.

Best

**Comment by igormatkovic** [comment]
> Take a look at #1066.
> This fix will be available in the next release.
> 
> Best

Yes, I can see that and my question was when will you tag the next release? 

I see this fix has been in there for about 2.5 months. 

Thanks

**Comment by gonzaloebiz** [comment]
We are testing the new release right now
No ETA, sorry

Best
